zerver/lib/push_notifications.py

```diff
@@ -1190,3 +1190,53 @@ def handle_push_notification(user_profile_id: int, missed_message: Dict[str, Any
     user_identity = UserPushIdentityCompat(user_id=user_profile.id)
     send_apple_push_notification(user_identity, apple_devices, apns_payload)
     send_android_push_notification(user_identity, android_devices, gcm_payload, gcm_options)
+
+
+def send_test_push_notification_directly_to_device(
+    user_identity: UserPushIdentityCompat,
+    device: DeviceToken,
+    base_payload: Dict[str, Any],
+    remote: Optional["RemoteZulipServer"] = None,
+) -> None:
+    payload = base_payload
+    payload["event"] = "test-by-device-token"
+
+    if device.kind == RemotePushDeviceToken.APNS:
+        apns_data = {
+            "alert": {
+                "title": "Test notification",
+                "body": "This is a test notification from <realm URL>.",
```

Comment:
These should be tagged for translation, assuming they are displayed to the user.